bit/count_ones.py

```diff
@@ -14,16 +14,58 @@
 
 Number of loops is
 equal to the number of 1s in the binary representation."""
+import unittest
 
-def count_ones(n):
+
+def count_ones_recur(n):
     """Using Brian Kernighan’s Algorithm. (Recursive Approach)"""
-    if not n: return 0
-    return 1 + count_ones(n & (n-1))
 
-def count_ones(n):
+    if not n:
+        return 0
+    return 1 + count_ones_recur(n & (n-1))
```

Comment:
I think it must be ```return 1 + count_ones(n & (n-1))``` because you renamed the function.

Reply:
`return 1 + count_ones(n & (n-1))` was the old one. Now recursive version is renamed to `count_ones_recur`, and hence `return 1 + count_ones_recur(n & (n-1))`

Reply:
@goswami-rahul  OK.